extensions/ql-vscode/src/qlpack-discovery.ts

```diff
@@ -16,18 +16,17 @@ export class QLPackDiscovery extends Discovery<QlpacksInfo> {
   private readonly watcher = this.push(new MultiFileSystemWatcher());
   private _qlPacks: readonly QLPack[] = [];
 
-  constructor(private readonly workspaceFolder: WorkspaceFolder,
-    private readonly cliServer: CodeQLCliServer) {
-
+  constructor(
+    private readonly workspaceFolder: WorkspaceFolder,
+    private readonly cliServer: CodeQLCliServer
+  ) {
     super();
 
     // Watch for any changes to `qlpack.yml` files in this workspace folder.
     // TODO: The CLI server should tell us what paths to watch for.
     this.watcher.addWatch(new RelativePattern(this.workspaceFolder, '**/qlpack.yml'));
     this.watcher.addWatch(new RelativePattern(this.workspaceFolder, '**/.codeqlmanifest.json'));
     this.push(this.watcher.onDidChange(this.handleQLPackFileChanged, this));
-
```

Comment:
Can you explain why factoring out these `.refresh()` calls out of the constructor is ok? Do we not need to call refresh when we first construct these for real, outside of tests?

Reply:
They are only constructed in one place and we refresh both of them there/ https://github.com/github/vscode-codeql/pull/518/files#diff-070c9f57c79a24599bedea1127e05a06R102-R103

If you prefer, I could stub out the `refresh` method for the tests and move it back to the way it was.